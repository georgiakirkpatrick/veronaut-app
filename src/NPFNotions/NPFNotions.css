.NPFNotions .FormPopUp .FormButton {
    background-color: #F2E7F1;
}