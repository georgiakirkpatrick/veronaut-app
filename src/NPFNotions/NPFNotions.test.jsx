import React from 'react'
import { render, screen } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import NPFNotions from './NPFNotions'

describe('NPFNotions', () => {
  it ('renders without crashing', () => {
    render(<NPFNotions />)
    screen.debug()
  })

  it ('render the UI as expected', () => {
    const tree = render(<NPFNotions />)
    expect(tree).toMatchSnapshot()
  })
})