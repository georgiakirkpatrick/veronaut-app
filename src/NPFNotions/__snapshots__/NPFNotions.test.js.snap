// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NPFNotions render the UI as expected 1`] = `
<div
  className="NPFNotions"
  id="notions"
>
  <section
    className="FormPage"
  >
    <h1>
      Notions
    </h1>
    <form
      autoComplete="off"
      id="form-page"
    >
      <div
        className="FormPromptWithSub"
      >
        <p
          className="FormPromptWithSub__prompt"
        >
          If the product includes notions such as zippers or buttons, list them here.
        </p>
        <p
          className="FormPromptWithSub__subtitle"
        >
          If the product does not include notions, skip this section.
        </p>
      </div>
      <li>
        <fieldset
          className="FormFieldset"
        >
          <div />
          <button
            className="NewProductForm__remove"
            onClick={[Function]}
            type="button"
          >
            REMOVE
          </button>
          <div
            className="FormDropdown"
          >
            <label
              htmlFor="notion-type-0"
            >
              Notion type
            </label>
            
            <select
              name="typeId"
              onChange={[Function]}
              required={true}
              value={0}
            >
              <option
                id={0}
                value={0}
              >
                Select a notion type
              </option>
            </select>
          </div>
          <div
            className="FormPromptWithSub"
          >
            <p
              className="FormPromptWithSub__prompt"
            >
              
            </p>
            <p
              className="FormPromptWithSub__subtitle"
            >
              Notion type not listed?
            </p>
          </div>
          <button
            className="FormButton"
            onClick={[Function]}
            type="button"
          >
            Add a notion type
          </button>
          <section
            className="FormPopUp"
            id="notion-type-pop-up"
          >
            <button
              className="FormPopUp__close-button"
              onClick={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-times fa-w-11 fa-lg "
                data-icon="times"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 352 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </button>
            <header>
              <h1>
                Add a Notion Type
              </h1>
            </header>
            <div
              className="FormTextInput"
            >
              <label
                htmlFor="new-not-type"
              >
                Notion type name
              </label>
              <input
                id="new-not-type"
                name="newNotType"
                onChange={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
            <button
              className="FormButton"
              onClick={[Function]}
              type="button"
            >
              Submit notion type
            </button>
          </section>
          <div
            className="FormDropdown"
          >
            <label
              htmlFor="notion-location-0"
            >
              Location of notion manufacturing
            </label>
            
            <select
              name="countryId"
              onChange={[Function]}
              required={true}
              value={0}
            />
          </div>
          <div
            className="FormDropdown"
          >
            <label
              htmlFor="notion-factory-0"
            >
              Notion manufacturer
            </label>
            
            <select
              name="factoryId"
              onChange={[Function]}
              required={true}
              value={0}
            >
              <option
                id={0}
                value={0}
              >
                Select a manufacturer
              </option>
              <option
                id={1}
                value={1}
              >
                Not disclosed
              </option>
            </select>
          </div>
          <div
            className="FormPromptWithSub"
          >
            <p
              className="FormPromptWithSub__prompt"
            >
              
            </p>
            <p
              className="FormPromptWithSub__subtitle"
            >
              Manufacturer not listed?
            </p>
          </div>
          <button
            className="FormButton"
            onClick={[Function]}
            type="button"
          >
            Add a manufacturer
          </button>
          <section
            className="FormPopUp"
            id="notion-factory-pop-up"
          >
            <button
              className="FormPopUp__close-button"
              onClick={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-times fa-w-11 fa-lg "
                data-icon="times"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 352 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </button>
            <header>
              <h1>
                Add a Manufacturer
              </h1>
            </header>
            <div
              className="FormPromptWithSub"
            >
              <p
                className="FormPromptWithSub__prompt"
              >
                Add a notion manufacturer
              </p>
              <p
                className="FormPromptWithSub__subtitle"
              >
                
              </p>
            </div>
            <div
              className="FormTextInput"
            >
              <label
                htmlFor="notion-factory-name"
              >
                Factory name
              </label>
              <input
                id="notion-factory-name"
                name="name"
                onChange={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
            <div
              className="FormDropdown"
            >
              <label
                htmlFor="notion-factory-location"
              >
                Location
              </label>
              
              <select
                name="countryId"
                onChange={[Function]}
                required={true}
                value=""
              />
            </div>
            <div
              className="FormUrlInput"
            >
              <label
                htmlFor="notion-factory-website"
              >
                Website
              </label>
              <input
                id="notion-factory-website"
                name="website"
                onChange={[Function]}
                required={true}
                type="url"
                value=""
              />
            </div>
            <div
              className="FormTextInput"
            >
              <label
                htmlFor="notion-factory-notes"
              >
                Notes
              </label>
              <input
                id="notion-factory-notes"
                name="notes"
                onChange={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
            <button
              className="FormButton"
              onClick={[Function]}
              type="button"
            >
              Submit manufacturer
            </button>
          </section>
          <div
            className="FormDropdown"
          >
            <label
              htmlFor="notion-material-type-0"
            >
              Notion material
            </label>
            
            <select
              name="materialTypeId"
              onChange={[Function]}
              required={true}
              value={0}
            >
              <option
                id={0}
                value={0}
              >
                Select a fiber or material
              </option>
              <option
                id={1}
                value={1}
              >
                Not disclosed
              </option>
            </select>
          </div>
          <div
            className="FormPromptWithSub"
          >
            <p
              className="FormPromptWithSub__prompt"
            >
              
            </p>
            <p
              className="FormPromptWithSub__subtitle"
            >
              Material type not listed?
            </p>
          </div>
          <button
            className="FormButton"
            onClick={[Function]}
            type="button"
          >
            Add a material type
          </button>
          <section
            className="FormPopUp"
            id="new-material"
          >
            <button
              className="FormPopUp__close-button"
              onClick={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-times fa-w-11 fa-lg "
                data-icon="times"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 352 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </button>
            <header>
              <h1>
                Add a Material
              </h1>
            </header>
            <div
              className="FormTextInput"
            >
              <label
                htmlFor="new-material-name"
              >
                Material name
              </label>
              <input
                id="new-material-name"
                name="name"
                onChange={[Function]}
                required={true}
                type="text"
                value={1}
              />
            </div>
            <button
              className="FormButton"
              onClick={[Function]}
              type="button"
            >
              Submit material
            </button>
          </section>
          <div
            className="FormDropdown"
          >
            <label
              htmlFor="notion-material-origin-id-0"
            >
              Origin of notion material
            </label>
            
            <select
              name="materialOriginId"
              onChange={[Function]}
              required={true}
              value={0}
            />
          </div>
          <div
            className="FormDropdown"
          >
            <label
              htmlFor="notion-material-producer-id-0"
            >
              Notion material producer
            </label>
            
            <select
              name="materialProducerId"
              onChange={[Function]}
              required={true}
              value=""
            >
              <option
                id={0}
                value={0}
              >
                Select a producer
              </option>
              <option
                id={1}
                value={1}
              >
                Not disclosed
              </option>
            </select>
          </div>
          <div
            className="FormPromptWithSub"
          >
            <p
              className="FormPromptWithSub__prompt"
            >
              
            </p>
            <p
              className="FormPromptWithSub__subtitle"
            >
              Manufacturer not listed?
            </p>
          </div>
          <button
            className="FormButton"
            onClick={[Function]}
            type="button"
          >
            Add a manufacturer
          </button>
          <p>
            Does the notion have any of the following cerfications?
          </p>
          <div
            className="FormCheckboxSection"
          >
            <div
              className="empty"
            />
            <div
              className="FormCheckboxSection__options-container"
            />
          </div>
          <div
            className="FormPromptWithSub"
          >
            <p
              className="FormPromptWithSub__prompt"
            >
              
            </p>
            <p
              className="FormPromptWithSub__subtitle"
            >
              Certification not listed?
            </p>
          </div>
          <button
            className="FormButton"
            onClick={[Function]}
            type="button"
          >
            Add a certification
          </button>
          <section
            className="FormPopUp"
            id="not-cert-pop-up"
          >
            <button
              className="FormPopUp__close-button"
              onClick={[Function]}
              type="button"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-times fa-w-11 fa-lg "
                data-icon="times"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 352 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </button>
            <header>
              <h1>
                New Certification
              </h1>
            </header>
            <div
              className="FormPromptWithSub"
            >
              <p
                className="FormPromptWithSub__prompt"
              >
                Add a certification
              </p>
              <p
                className="FormPromptWithSub__subtitle"
              >
                
              </p>
            </div>
            <div
              className="FormTextInput"
            >
              <label
                htmlFor="not-cert-name"
              >
                Certification name
              </label>
              <input
                id="not-cert-name"
                name="name"
                onChange={[Function]}
                required={true}
                type="text"
                value=""
              />
            </div>
            <div
              className="FormUrlInput"
            >
              <label
                htmlFor="not-cert-website"
              >
                Website
              </label>
              <input
                id="not-cert-website"
                name="website"
                onChange={[Function]}
                required={true}
                type="url"
                value=""
              />
            </div>
            <button
              className="FormButton"
              onClick={[Function]}
              type="button"
            >
              Submit certification
            </button>
          </section>
          <div
            className="FormTextarea"
          >
            <label
              htmlFor="notes0"
            >
              If the brand offers additional information about notion materials or manufacturing, include them here.
            </label>
            <textarea
              id="notes0"
              name="notes"
              onChange={[Function]}
              required={true}
              rows="1"
              value=""
            />
          </div>
        </fieldset>
      </li>
      <button
        className="FormButton"
        onClick={[Function]}
        type="button"
      >
        Add a notion
      </button>
    </form>
    <div>
      
    </div>
  </section>
  <footer
    className="NPFFooter"
  >
    <button
      className="NPFFooter__previous"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-left fa-w-8 NPFFooter__left-arrow"
        data-icon="angle-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        PREVIOUS
      </span>
    </button>
    <button
      className="NPFFooter__next"
      onClick={[Function]}
      type="button"
    >
      <span>
        NEXT
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-right fa-w-8 NPFFooter__right-arrow"
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </footer>
</div>
`;
